tests/Feature/GraphQL/FilterTest.php

```diff
@@ -500,4 +512,490 @@ public function testProhibitsMultipleFieldsInSingleFilterObject(): void
+                        ],
+                        [
+                            'node' => [
+                                'name' => $this->projects['public2']->name,
+                            ],
+                        ],
+                    ],
+                ],
+            ],
+        ], true);
+    }
+
+    public function testFilterByAllMultipleRelationships(): void
+    {
+        $this->sites['site1'] = $this->makeSite();
+
+        $build1uuid = Str::uuid()->toString();
+        $this->projects['public1']->builds()->create([
+            'name' => 'build1',
+            'uuid' => $build1uuid,
+        ]);
+        $build2uuid = Str::uuid()->toString();
+        $this->projects['public1']->builds()->create([
+            'name' => 'build2',
+            'uuid' => $build2uuid,
+            'siteid' => $this->sites['site1']->id,
+        ]);
+
+        $this->actingAs($this->users['admin'])->graphQL('
+            query($uuid: String!, $siteid: ID!) {
+                projects(filters: {
+                    all: [
+                        {
+                            has: {
+                                builds: {
+                                    eq: {
+                                        uuid: $uuid
+                                    }
+                                }
+                            }
+                        },
+                        {
+                            has: {
+                                sites: {
+                                    eq: {
+                                        id: $siteid
+                                    }
+                                }
+                            }
+                        },
+                    ]
+                }) {
+                    edges {
+                        node {
+                            name
+                        }
+                    }
+                }
+            }
+        ', [
+            'uuid' => $build1uuid,
+            'siteid' => $this->sites['site1']->id,
+        ])->assertJson([
+            'data' => [
+                'projects' => [
+                    'edges' => [
+                        [
+                            'node' => [
+                                'name' => $this->projects['public1']->name,
+                            ],
+                        ],
+                    ],
+                ],
+            ],
+        ], true);
+    }
+
+    public function testFilterByRelationshipsOfRelationships(): void
+    {
+        $build1uuid = Str::uuid()->toString();
+        $target1name = Str::uuid()->toString();
+        $this->projects['public1']->builds()->create([
+            'name' => 'build1',
+            'uuid' => $build1uuid,
+        ])->targets()->create([
+            'name' => $target1name,
+            'type' => TargetType::UNKNOWN,
+        ]);
+
+        $build2uuid = Str::uuid()->toString();
+        $target2name = Str::uuid()->toString();
```

Comment:
What does adding this second build and target do for this test?

Reply:
Ah, good catch.  This build is supposed to be attached to a different project (e.g., `public2`).